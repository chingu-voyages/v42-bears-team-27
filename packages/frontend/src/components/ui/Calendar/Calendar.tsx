/* eslint-disable react/jsx-props-no-spreading */
import type { RefAttributes, ForwardRefExoticComponent } from 'react';
import { forwardRef } from 'react';
import type { ThemeUIStyleObject } from 'theme-ui';
import ReactCalendar, {
  CalendarProps as ReactCalenderProps,
} from 'react-calendar';
import 'react-calendar/dist/Calendar.css';

export interface CalendarProps extends ReactCalenderProps {
  sx?: ThemeUIStyleObject;
}

const resetStyles: ThemeUIStyleObject = {
  variant: 'text.link',
  borderColor: 'gray',
  '.react-calendar__month-view__weekdays': {
    textTransform: 'none',
    '& abbr': {
      textDecoration: 'none',
    },
  },
  '.react-calendar__month-view__days__day': {
    color: 'primary',
  },
  '.react-calendar__month-view__days__day--neighboringMonth': {
    color: 'gray',
  },
  '.react-calendar__month-view__days__day--weekend': {
    color: '#ff5c58',
  },
};

export const Calendar: ForwardRefExoticComponent<
CalendarProps & RefAttributes<HTMLDivElement>
> = forwardRef(({ sx, ...rest }, ref) => (
  <ReactCalendar ref={ref} sx={{ ...resetStyles, ...sx }} {...rest} />
));

if (process.env.NODE_ENV !== 'production') {
  Calendar.displayName = 'Calendar';
}
