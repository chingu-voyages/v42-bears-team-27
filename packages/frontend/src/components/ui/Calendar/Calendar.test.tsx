import { render } from 'test-utils';

import { Calendar } from './Calendar';

describe('<Calendar />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Calendar />);
      expect(container).toMatchSnapshot();
    });
  });
});
