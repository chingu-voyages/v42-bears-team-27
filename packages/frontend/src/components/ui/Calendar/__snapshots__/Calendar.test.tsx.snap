// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> Snapshots should match snapshot 1`] = `
.emotion-0 {
  font-family: "Nunito",sans-serif;
  font-size: 18px;
  font-weight: 600;
  line-height: auto;
  border-color: var(--theme-ui-colors-gray);
}

.emotion-0 .react-calendar__month-view__weekdays {
  text-transform: none;
}

.emotion-0 .react-calendar__month-view__weekdays abbr {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 .react-calendar__month-view__days__day {
  color: var(--theme-ui-colors-primary);
}

.emotion-0 .react-calendar__month-view__days__day--neighboringMonth {
  color: var(--theme-ui-colors-gray);
}

.emotion-0 .react-calendar__month-view__days__day--weekend {
  color: #ff5c58;
}

<div>
  <div
    class="react-calendar emotion-0"
  >
    <div
      class="react-calendar__navigation"
    >
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
        type="button"
      >
        «
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
        type="button"
      >
        ‹
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__label"
        style="flex-grow: 1;"
        type="button"
      >
        <span
          class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
        >
          January 2023
        </span>
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
        type="button"
      >
        ›
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
        type="button"
      >
        »
      </button>
    </div>
    <div
      class="react-calendar__viewContainer"
    >
      <div
        class="react-calendar__month-view"
      >
        <div
          style="display: flex; align-items: flex-end;"
        >
          <div
            style="flex-grow: 1; width: 100%;"
          >
            <div
              class="react-calendar__month-view__weekdays"
              style="display: flex; flex-wrap: no-wrap;"
            >
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Monday"
                  title="Monday"
                >
                  Mon
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Tuesday"
                  title="Tuesday"
                >
                  Tue
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Wednesday"
                  title="Wednesday"
                >
                  Wed
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Thursday"
                  title="Thursday"
                >
                  Thu
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Friday"
                  title="Friday"
                >
                  Fri
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Saturday"
                  title="Saturday"
                >
                  Sat
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
              >
                <abbr
                  aria-label="Sunday"
                  title="Sunday"
                >
                  Sun
                </abbr>
              </div>
            </div>
            <div
              class="react-calendar__month-view__days"
              style="display: flex; flex-wrap: wrap;"
            >
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 26, 2022"
                >
                  26
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 27, 2022"
                >
                  27
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 28, 2022"
                >
                  28
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 29, 2022"
                >
                  29
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 30, 2022"
                >
                  30
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 31, 2022"
                >
                  31
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 1, 2023"
                >
                  1
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 2, 2023"
                >
                  2
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 3, 2023"
                >
                  3
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 4, 2023"
                >
                  4
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 5, 2023"
                >
                  5
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 6, 2023"
                >
                  6
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 7, 2023"
                >
                  7
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 8, 2023"
                >
                  8
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 9, 2023"
                >
                  9
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 10, 2023"
                >
                  10
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 11, 2023"
                >
                  11
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 12, 2023"
                >
                  12
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 13, 2023"
                >
                  13
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 14, 2023"
                >
                  14
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 15, 2023"
                >
                  15
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 16, 2023"
                >
                  16
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 17, 2023"
                >
                  17
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 18, 2023"
                >
                  18
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 19, 2023"
                >
                  19
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 20, 2023"
                >
                  20
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 21, 2023"
                >
                  21
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 22, 2023"
                >
                  22
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 23, 2023"
                >
                  23
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 24, 2023"
                >
                  24
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 25, 2023"
                >
                  25
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 26, 2023"
                >
                  26
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 27, 2023"
                >
                  27
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 28, 2023"
                >
                  28
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 29, 2023"
                >
                  29
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 30, 2023"
                >
                  30
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 31, 2023"
                >
                  31
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 1, 2023"
                >
                  1
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 2, 2023"
                >
                  2
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 3, 2023"
                >
                  3
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 4, 2023"
                >
                  4
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 5, 2023"
                >
                  5
                </abbr>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
